tuist/tuist
# Improve app extensions autogenerated schemes

Resolves: https://github.com/tuist/tuist/issues/1494

### Short description 📝

The autogenerated schemes for app extensions do not launch the corresponding extension.

### Solution 📦

Update the auto generated schemes to account for extensions. For most of the extensions I tested, the pattern was to add the host app as the runnable target.

### Implementation 👩‍💻👨‍💻
- [x] Update autogenerated scheme mapper:
   -  Add the extension host app as the runnable target
   - Add the extension host app to the list of buildable targets
- [x] Update scheme generator to set the appropriate settings for extensions
   - The scheme is marked with `wasCreatedForAppExtension`
   - Launch subtype 2 is used
   - The "Ask on Launch" option is selected
   - The launcher identifier is `Xcode.IDEFoundation.Launcher.PosixSpawn`
   - The debugger identifier is now blank
   - The mechanism for determining if a scheme is for an app extension is via convention, if the first buildable target is an app extension.
- [x] Update the change log

### Test Plan 🛠

- Run `swift run tuist generate` within `fixtures/ios_app_with_extensions`
- Open the generated project
- Select and run MessageExtension scheme
- Verify the message extension is launched in the simulator
- On Xcode 12, select and run the Widget scheme
- Verify the Widget is launched in the simulator
